guides/source/active_storage_overview.md

```diff
@@ -960,75 +960,184 @@ class Uploader {
 
 NOTE: Using [Direct Uploads](#direct-uploads) can sometimes result in a file that uploads, but never attaches to a record. Consider [purging unattached uploads](#purging-unattached-uploads).
 
-Discarding Files Stored During System Tests
+Testing
 -------------------------------------------
 
-System tests clean up test data by rolling back a transaction. Because destroy
+Use [`fixture_file_upload`][] to test uploading a file in an integration or controller test.
+Rails handles files like any other parameter.
+
+```ruby
+class SignupController < ActionDispatch::IntegrationTest
+  test "can sign up" do
+    post signup_path, params: {
+      name: "David",
+      avatar: fixture_file_upload("david.png", "image/png")
+    }
+
+    user = User.order(:created_at).last
+    assert user.avatar.attached?
+  end
+end
+```
+
+[`fixture_file_upload`]: https://api.rubyonrails.org/classes/ActionDispatch/TestProcess/FixtureFile.html
+
+### Discarding files created during tests
+
+#### System tests
+
+System tests clean up test data by rolling back a transaction. Because `destroy`
 is never called on an object, the attached files are never cleaned up. If you
 want to clear the files, you can do it in an `after_teardown` callback. Doing it
 here ensures that all connections created during the test are complete and
 you won't receive an error from Active Storage saying it can't find a file.
 
 ```ruby
-class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
+class ActionDispatch::SystemTestCase
```

Comment:
Sorry, I just noticed this. `ApplicationSystemTestCase` defines `driven_by` and is generated if you don't already have an app level system test case (`bin/rails g system_test xyz`). Can you explain why this was refactored to patch `ActionDispatch::SystemTestCase` directly?

Reply:
Technically `ActionDispatch::SystemTestCase` defines `driven_by`, but point taken about the generator - I didn't know about that. I changed it to make the examples more consistent; I'll change the system ones back.

Reply:
Er, yes. My mistake, I meant `driven_by` is usually _called_ on the application test case.

Will merge now. Thanks for your patience on this!